File: crio/python/ds/List/ListNode.py
class ListNode:
	def __init__(self, x):
		self.val = x
		self.next = None

def createList(numbers):
	if(len(numbers) == 0):
		return None

	head = ListNode(numbers[0])
	node = head
	for i in range(1, len(numbers)):
		node.next = ListNode(numbers[i])
		node = node.next

	return head

def extractList(head):
	arr = []

	while(head):
		arr.append(head.val)
		head = head.next

	return arr

def generateLinkedListWithCycle(numbers, pos):
	List = createList(numbers)
	if pos == -1:
		return List

	head = List
	tail = List
	while pos > 0:
		pos -= 1
		head = head.next
	while(tail.next):
		tail = tail.next
	tail.next = head
	return List

File: crio/java/ds/List/ListNode.java
package crio.ds.List;

import java.io.*;
import java.util.*;

public class ListNode {
    public int val;
    public ListNode next;
    public ListNode(int x) { val = x; next = null; }

    public static ListNode createList(List<Integer> numbers) {
        if (numbers.size() == 0)
            return null;

        ListNode head = new ListNode(numbers.get(0));
        ListNode node = head;

        for (int i = 1; i < numbers.size(); ++i) {
            node.next = new ListNode(numbers.get(i));
            node = node.next;
        }

        return head;
    }

    public static List<Integer> extractList(ListNode head) {
        List<Integer> lst = new ArrayList<Integer>();

        while(head != null) {
            lst.add(head.val);
            head = head.next;
        }

        return lst;
    }
}
